### PR TITLE
update addons for k8s 1.8

### PR DESCRIPTION
**Pull request by moelsayed**

### CODE REVIEW

**Comment by moelsayed** [comment]
@alena1108 Validation tests passed. Also check services seem to be working properly.

File: addon-templates/kubectl/dashboard/dashboard-controller.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ metadata:
   labels:
     k8s-app: kubernetes-dashboard
     kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
```
Comment (galal-hussein) [inline addon-templates/kubectl/dashboard/dashboard-controller.yaml]: The addonmanager labels are needed when addon manager component is deployed to ensure the availability of the addons, can you remove this label?

File: addon-templates/kubectl/dashboard/dashboard-controller.yaml [hunk 2]
```diff
@@ -16,20 +17,19 @@ spec:
         k8s-app: kubernetes-dashboard
       annotations:
         scheduler.alpha.kubernetes.io/critical-pod: ''
-        scheduler.alpha.kubernetes.io/tolerations: '[{"key":"CriticalAddonsOnly", "operator":"Exists"}]'
```
Comment (galal-hussein) [inline addon-templates/kubectl/dashboard/dashboard-controller.yaml]: marking the addons as critical is important and to make sure its always scheduled.

File: addon-templates/kubectl/heapster/grafana-deployment.yaml [hunk 3]
```diff
@@ -13,11 +13,14 @@ spec:
     spec:
       containers:
       - name: grafana
-        image: $GCR_IO_REGISTRY/google_containers/heapster-grafana-amd64:v4.0.2
+        image: $GCR_IO_REGISTRY/google_containers/heapster-grafana-amd64:v4.4.3
         ports:
           - containerPort: 3000
             protocol: TCP
         volumeMounts:
+        - mountPath: /etc/ssl/certs
```
Comment (galal-hussein) [inline addon-templates/kubectl/heapster/grafana-deployment.yaml]: is the ssl certs needed, AFAIK we don't use TLS with grafana endpoint

File: addon-templates/kubectl/heapster/grafana-deployment.yaml [hunk 4]
```diff
@@ -43,5 +43,8 @@ spec:
           value: /
       serviceAccountName: io-rancher-system
       volumes:
+      - name: ca-certificates
```
Comment (galal-hussein) [inline addon-templates/kubectl/heapster/grafana-deployment.yaml]: The volumes are still defined here, this should be removed too.

**Review by galal-hussein — APPROVED** [review]
LGTM, thanks

**Pull request merged**